components/eam/src/physics/cam/zm_transport.F90

```diff
@@ -0,0 +1,589 @@
+               do k = jt(i),mx(i)
+                  if (dcondt(i,k)*dt+const(i,k)<0._r8) then
+                     negadt = dcondt(i,k)+const(i,k)/dt
+                     dcondt(i,k) = -const(i,k)/dt
+                     do kk= k+1, mx(i)
+                        if (negadt<0._r8 .and. dcondt(i,kk)*dt+const(i,kk)>0._r8 ) then
+                           qtmp = dcondt(i,kk)+negadt*dptmp(i,k)/dptmp(i,kk)
+                           if (qtmp*dt+const(i,kk)>0._r8) then
+                              dcondt(i,kk)= qtmp
+                              negadt=0._r8
+                           else
+                              negadt= negadt+(const(i,kk)/dt+dcondt(i,kk))*dptmp(i,kk)/dptmp(i,k)
+                              dcondt(i,kk)= -const(i,kk)/dt
+                           end if
+                        end if
+                     end do
+                     do kk= k-1, jt(i), -1
+                        if (negadt<0._r8 .and. dcondt(i,kk)*dt+const(i,kk)>0._r8 ) then
+                           qtmp = dcondt(i,kk)+negadt*dptmp(i,k)/dptmp(i,kk)
+                           if (qtmp*dt+const(i,kk)>0._r8) then
+                              dcondt(i,kk)= qtmp
+                              negadt=0._r8
+                           else
+                              negadt= negadt+(const(i,kk)/dt+dcondt(i,kk))*dptmp(i,kk)/dptmp(i,k)
+                              dcondt(i,kk)= -const(i,kk)/dt
+                           end if
+                        end if
+                     end do
+                     if (negadt<0._r8) then
+                        dcondt(i,k) = dcondt(i,k) - negadt
+                     end if
+                  end if
+               end do
+            end do
+         end if
+
+         ! Initialize output tendency to zero, then scatter tendency back to full array
+         dqdt(:,:,m) = 0._r8
+         do k = 1,pver
+            kp1 = min(pver,k+1)
+#ifdef CPRCRAY
+!DIR$ CONCURRENT
+#endif
+            do i = il1g,il2g
+               dqdt(ideep(i),k,m) = dcondt(i,k)
+            end do
+         end do
+
+      end if ! for doconvtran
+
+   end do ! m = 2, ncnst
+
+   return
+
+end subroutine zm_transport_tracer
+
+!===================================================================================================
+
+subroutine zm_transport_momentum( lchnk, ncol, wind_in, nwind, &
+                                  mu, md, du, eu, ed, dp, &
+                                  jt, mx, ideep, il1g, il2g, &
+                                  wind_tend, pguall, pgdall, icwu, icwd, dt, seten )
+   !---------------------------------------------------------------------------- 
+   ! Purpose: Convective transport of momentum
+   !----------------------------------------------------------------------------
+   implicit none
+   !----------------------------------------------------------------------------
+   ! Arguments
+   integer,                               intent(in)  :: lchnk       ! chunk identifier
```

Comment:
Do any of these routines use `lchnk`?  I did a quick search and couldn't find any reference to it other than having it be declared.

Reply:
Good catch! I think this and some other variables were just added to all ZM subroutines by default in case they were needed, but I can't think of a good justification for leaving this in, so I'll remove it from the interface.

Reply:
fixed.